### ISSUE TITLE
Incorrect Objects Created When Referencing Embedded Custom Messages

### ISSUE THREAD

**Issue by buckhx** [improbable-eng/ts-protoc-gen#7]
The actual proto defs I'm using are more complex than this, but the issue boils down to the state of the following pseudo-protos

a.proto
```
message A {
    message Embed {
        string msg = 1;
    }
}
```

b.proto
```
import 'file_a.proto';
message B {
    message C {
        A.Embed one = 1;
    }
}
```

The objects generated for C reference Embed directly without prefixing A.Embed

Here is an example of what is currently generated
```
proto.package.b.B.C.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }    
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new a_pb.Embed;
      reader.readMessage(value,a_pb.Embed.deserializeBinaryFromReader);
      msg.setOne(value);
      break;
    default:
      reader.skipField();
      break;
    }    
  }
  return msg; 
};
```

And what it should look like with A.Embed
```
proto.package.b.B.C.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }    
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new a_pb.A.Embed;
      reader.readMessage(value,a_pb.A.Embed.deserializeBinaryFromReader);
      msg.setOne(value);
      break;
    default:
      reader.skipField();
      break;
    }    
  }
  return msg; 
};
```
I've currently hacked a solution to sed replace a_pb.Embed w/ a_pb.A.Embed. Happy to PR if pointed in the right direction as well. The generation code can be a bit hard to follow.

**Comment by MarcusLongmuir** [comment]
Hi @buckhx 

Apologies for the long delay. The `.js` output that you're seeing the error in is the output from [protoc's `--js_out`](https://github.com/google/protobuf/tree/master/js). You're best [raising an issue there](https://github.com/google/protobuf/issues?q=is%3Aopen+is%3Aissue+label%3Ajavascript) if you're still seeing this behaviour.
